ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -937,6 +998,13 @@ private void initialise ()
      */
     private WorkerService _workerService = null;
 
+    /**
+     * flag to signal that there are transactions to recover;
+     * this flag could ONLY be set to false during the
+     * periodic recovery cycle (i.e. doWorkInternal()).
+     */
+    private volatile boolean _blockSuspension = true;
```

Comment:
I need to think some more about the safety of this flag but I think my review should give you enough "food for thought" while I think about that.

Reply:
This has been discussed in-person. Can we resolve this conversation or is there anything I missed?